srv/modules/runners/populate.py

```diff
@@ -1009,30 +1006,31 @@ def _get_existing_cluster_network(addrs, public_network=None):
                 if "inet" in data:
                     for inet_data in data["inet"]:
                         if i == "0.0.0.0":
-                            # If running on 0.0.0.0, assume we can use public_network
-                            if public_network:
-                                ip = ipaddress.ip_interface(u"{}".format(public_network))
+                            # If running on 0.0.0.0, assume we can use public_networks
+                            if public_networks:
+                                for n in public_networks:
+                                    ip = ipaddress.ip_interface(u"{}".format(n))
+                                    minion_networks.append(str(ip.network))
                             else:
-                                ip = ipaddress.ip_interface(u"{}/{}".format(i,
-                                                                            inet_data["netmask"]))
-                            minion_networks.append(str(ip.network))
+                                ip = ipaddress.ip_interface(u"{}/{}".format(i, inet_data["netmask"]))
+                                minion_networks.append(str(ip.network))
                         elif inet_data["address"] == i:
                             ip = ipaddress.ip_interface(u"{}/{}".format(inet_data["address"],
                                                                         inet_data["netmask"]))
                             minion_networks.append(str(ip.network))
 
-    # Check for consistency across all entries.
-    if len(set(minion_networks)) == 1:
-        # We have equal entries.
-        network = minion_networks[0]
-    else:
-        # We have multiple possible networks.  This is liable to happen with OSDs
-        # when there is a private cluster network.  Let's try to remove the public
-        # network.
-        minion_networks = [n for n in minion_networks if n != public_network]
-        network = minion_networks[0] if len(set(minion_networks)) == 1 else None
+    # Collapse minion_networks back to unique items.  Usually there'll
+    # be only one network after this, unless the cluster is using a
+    # separate public and private network, or is using multiple public
+    # and/or private networks
+    minion_networks = list(set(minion_networks))
```

Comment:
Any reason you're not using ```minion_networks = set()``` right away?

Reply:
Just that when I wrote it I was thinking "minion_networks" needs to always be a list. I now realise that the `[n for n in minion_networks]` business on the next line would work just fine against a set, and would return a list, so `list()` in the above line is redundant.